lib/rules/no-lone-blocks.js

```diff
@@ -11,17 +13,90 @@
 
 module.exports = function(context) {
 
-    return {
-        "BlockStatement": function (node) {
-            // Check for any occurrence of BlockStatement > BlockStatement or
-            // Program > BlockStatement
-            var parent = context.getAncestors().pop();
-            if (parent.type === "BlockStatement" || parent.type === "Program") {
-                context.report(node, "Block is nested inside another block.");
+    // A stack of lone blocks to be checked for block-level bindings
+    var loneBlocks = [],
+        ruleDef;
+
+    /**
+     * Reports a node as invalid.
+     * @param {ASTNode} node - The node to be reported.
+     * @returns {void}
+    */
+    function report(node) {
+        context.report(node, "Nested block is redundant.");
```

Comment:
I think "Block is redundant", or maybe "Block is unnecessary". I think either one is fine, and clearer then original message.